### ISSUE TITLE
Prefix method is missing in v2.0.0 release

### ISSUE THREAD

**Issue by kenota** [asdine/storm#196]
Db.Prefix() method as described in Readme.md is missing in v2.0.0 release

**Comment by asdine** [comment]
@kenota Thank you 👍

**Comment by kenota** [comment]
Wow, that was so quick, thank you @asdine !

Tested, working fine. Just one note: it seems that having an index is required for Prefix to work, if there is no index it will fail. Is it expected behaviour? I looked at original PR and i think idea was to fallback to using regexp scan.

If it is designed behaviour (I assume it makes sense to enforce index if you are doing prefix scan anyway), probably worth mentioning it in the docs.

**Comment by asdine** [comment]
It's not the intended behaviour, actually it should work with or without indexes.
Can you provide a working example please so i can investigate on this issue ?

**Comment by kenota** [comment]
@asdine 
This test case is failing for me on 2.0.1 unless i create index:

```go
package main

import (
	"io/ioutil"
	"os"
	"testing"

	"github.com/asdine/storm"
	"github.com/stretchr/testify/require"
)

type Entity struct {
	ID string `storm:"id"`
}

func TestPrefixScan(t *testing.T) {
	f, err := ioutil.TempFile("", "test")
	require.Nil(t, err)
	defer os.Remove(f.Name())

	db, err := storm.Open(f.Name())
	require.Nil(t, err)

	require.Nil(t, db.Init(&Entity{}))

	require.Nil(t, db.Save(&Entity{ID: "1"}))
	require.Nil(t, db.Save(&Entity{ID: "12"}))

	var res []*Entity
        // Getting  Expected nil, but got: &errors.errorString{s:"index not found"} here
	require.Nil(t, db.Prefix("ID", "1", &res))

}
```

**Comment by asdine** [comment]
Thank you, i have found the bug it was specific to ID fields. I have pushed this fix 68fc73b635f890fe7ba2f3b15ce80c85b28a744f, @kenota can you confirm that it works for you ?

**Comment by kenota** [comment]
@asdine Yep! Can confirm that my original code is now working without index. Thank you! 👍
